menus/pause_menu.gd

```diff
@@ -1,11 +1,28 @@
 extends Control
 
+var game_paused : bool = false:
+	get:
+		return game_paused
+	set(value):
+		game_paused = value
+		get_tree().paused = game_paused
+		_update_visibility()
 
-# Called when the node enters the scene tree for the first time.
 func _ready():
-	pass # Replace with function body.
+	hide()
 
+func _input(event : InputEvent):
+	if(event.is_action_pressed("ui_cancel")):
+		game_paused = !game_paused
+
+func _update_visibility():
+	if game_paused:
+		show()
+	else:
+		hide()
 
 func _on_quit_button_pressed():
-	pass
-	
+	get_tree().quit()
+
+func _on_resume_button_pressed():
```

Comment:
Pls move this function to above `_on_quit_button_pressed`, just because in the UI the resume it's before the quit 😅